java/daikon/DynComp.java

```diff
@@ -300,21 +281,23 @@ void start_target(String premain_args, String[] target_args) {
             "See the Daikon manual, section \"Instrumenting the JDK with DynComp\" for help.%n");
         System.exit(1);
       }
-    }
-
-    // For some unknown reason, groovy cannot be on the boot classpath.
-    String[] cpaths = cp.split(File.pathSeparator);
-    for (int i = 0; i < cpaths.length; i++) {
-      if (cpaths[i].contains("groovy")) {
-        String[] path_elements = cpaths[i].split(Pattern.quote(File.separator));
-        if (path_elements[path_elements.length - 1].matches("^groovy.*\\.jar$")) {
-          cpaths[i] = "";
+      // We need to add the location of Daikon to the boot classpath. We
+      // do this by inspecting each element of the classpath for either:
+      // a jar file that contains "DynComp.class"
+      // a path that ends in "java/lib/<something>.jar
+      // a path that contains "daikon/DynComp.class"
+      //
+      // If any of thses cases are true, we append the path to the boot
+      // classpath. We then repeat the process with the next element of
+      // the classpath.
+      for (String path : cp.split(File.pathSeparator)) {
+        if (isDaikonOnPath(path)) {
+          daikonPath = daikonPath + File.pathSeparator + path;
         }
       }
+      // debuging
+      System.out.println("daikonPath: " + daikonPath);
```

Comment:
This should be removed or protected by a flag.

Reply:
cahnged to basic.log ...